ado_downloader/__main__.py

```diff
@@ -445,6 +445,7 @@ def select_pipeline_run(
 ) -> int:
     az_command = shutil.which("az")
     pipeline = artifact_sources["pipelines"][pipeline_name]
+    allowed_build_reasons = artifact_sources["allowedBuildReasons"]
```

Comment:
Could we add a reasonable default here, in case this isn't specified in the YAML file? I could see an argument either for (a) defaulting this to the previous hard-coded list for any existing uses, or (b) defaulting this such that, if left unspecified, build reasons will not be used as a filter.

Reply:
Please correct me if I am mistaken. If no list is provided in the yaml file, the first default would evaluate `br["reason"] not in allowed_build_reasons` and may or may not return False early. The second one would skip the if statement. So the behavior could be different based on which route we take. Which one is preferred?

Reply:
Actually, if no list is provided in the YAML file, then I'd expect an exception to get thrown by the retrieve on line 448. `artifact_sources.get("allowedBuildReasons")` is an alternative retrieval method that would return `None` or a legitimate value retrieved from the YAML file.

And yes, I was identifying two valid but distinct options in my response. I could have been clearer, but I was asking for your recommendation. I think either option is defensible. Upon further reflection, I think it's fair to only apply build reason as a filter if `allowedBuildReasons` is specified in YAML.